src/mito/src/table.rs

```diff
@@ -529,9 +516,44 @@ impl<R: Region> MitoTable<R> {
         Ok(table_info)
     }
 
+    /// Remove regions
+    /// Notes: Please release regions in StorageEngine.
+    pub async fn remove_regions(&self, region_numbers: &[RegionNumber]) -> TableResult<()> {
+        self.regions.rcu(|regions| {
+            let mut regions = HashMap::clone(regions);
+            for region in region_numbers {
+                regions.remove(region);
+            }
+
+            Arc::new(regions)
+        });
+
+        Ok(())
+    }
+
+    pub async fn drop_regions(&self) -> TableResult<()> {
+        let regions = self.regions.load();
+        futures::future::try_join_all(regions.values().map(|region| region.drop_region()))
+            .await
+            .map_err(BoxedError::new)
+            .context(table_error::TableOperationSnafu)?;
+
+        let regions = regions.iter().map(|(k, _)| *k).collect::<Vec<_>>();
+
+        self.remove_regions(&regions).await?;
```

Comment:
It seems `remove_regions` should be called before dropping regions? Otherwise the regions could be read while dropping.